### PR TITLE
BUG/API: SeriesGroupBy reduction with numeric_only=True

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

This addresses the SeriesGroupBy (easier) half of #41291:

1) When we have numeric_only=True and non-numeric data in a SeriesGroupBy reduction, raise NotImplementedError (matching Series behavior) instead of falling back to an often-wrong fallback (xref #41341)
2) When the user doesn't explicitly pass `numeric_only=True`, change the default to False for SeriesGroupBy, leaving DataFrameGroupBy unaffected.

### CODE REVIEW

**Comment by jreback** [comment]
if you can rebase this one

**Comment by jreback** [comment]
haven't looked yet

**Comment by jbrockmendel** [comment]
Rebased+green

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -1099,6 +1099,24 @@ def _wrap_transformed_output(self, output: Mapping[base.OutputKey, ArrayLike]):
     def _wrap_applied_output(self, data, keys, values, not_indexed_same: bool = False):
         raise AbstractMethodError(self)
 
+    def _resolve_numeric_only(self, numeric_only: bool | lib.NoDefault) -> bool:
+        """
+        For SeriesGroupBy we want the default to be False (to match Series behavior).
+        For DataFrameGroupBy we want it to be True (for backwards-compat).
+        """
+        # GH#41291
+        if numeric_only is lib.no_default:
```
Comment (jreback) [inline pandas/core/groupby/groupby.py:1108]: are we deprecating the dataframe groupby behavior here?

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -1822,11 +1830,36 @@ def test_empty_groupby(columns, keys, values, method, op, request):
     df = df.iloc[:0]
 
     gb = df.groupby(keys)[columns]
-    if method == "attr":
-        result = getattr(gb, op)()
-    else:
-        result = getattr(gb, method)(op)
 
+    def get_result():
+        if method == "attr":
```
Comment (jreback) [inline pandas/tests/groupby/test_groupby.py:1835]: this is a super complicated test. can you try to split in followons.

File: doc/source/whatsnew/v1.3.0.rst [hunk 3]
```diff
@@ -906,6 +906,7 @@ Groupby/resample/rolling
 - Bug in :meth:`DataFrameGroupBy.__getitem__` with non-unique columns incorrectly returning a malformed :class:`SeriesGroupBy` instead of :class:`DataFrameGroupBy` (:issue:`41427`)
 - Bug in :meth:`DataFrameGroupBy.transform` with non-unique columns incorrectly raising ``AttributeError`` (:issue:`41427`)
 - Bug in :meth:`Resampler.apply` with non-unique columns incorrectly dropping duplicated columns (:issue:`41445`)
+- Bug in :meth:`SeriesGroupBy` aggregations incorrectly returning empty :class:`Series` instead of raising ``TypeError`` on aggregations that are invalid for its dtype (:issue:`41342`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:909]: can you give an example here that a user would understand

**Review by jreback — CHANGES_REQUESTED** [review]
can you rebase and a couple of comments

File: pandas/core/groupby/groupby.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/groupby.py:1108]: no, this is just a way of having the effective default be different for SeriesGroupBy vs DataFrameGroupBy

File: pandas/tests/groupby/test_groupby.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/groupby/test_groupby.py:1835]: agreed

File: doc/source/whatsnew/v1.3.0.rst (hunk 3 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.3.0.rst]: updated + green

File: pandas/core/groupby/groupby.py [hunk 4]
```diff
@@ -1099,6 +1099,24 @@ def _wrap_transformed_output(self, output: Mapping[base.OutputKey, ArrayLike]):
     def _wrap_applied_output(self, data, keys, values, not_indexed_same: bool = False):
         raise AbstractMethodError(self)
 
+    def _resolve_numeric_only(self, numeric_only: bool | lib.NoDefault) -> bool:
+        """
+        For SeriesGroupBy we want the default to be False (to match Series behavior).
+        For DataFrameGroupBy we want it to be True (for backwards-compat).
```
Comment (rhshadrach) [inline pandas/core/groupby/groupby.py:1105]: Can you make this a proper docstring: one line short description and a Parameters section.

**Review by rhshadrach — CHANGES_REQUESTED** [review]

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by jbrockmendel** [comment]
@jreback i think comments have been addressed; this is a blocker for fixing the analogous DataFrameGroupBy behavior

**Review by jreback — APPROVED** [review]
lgtm. can merge on green.

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**